src/RetireNet.Packages.Tool/Services/RetireLogger.cs

```diff
@@ -94,27 +115,57 @@ internal void LogPackagesToRetireInternal()
+                var nugetReferences = _nugetReferenceService.GetNugetReferences(lockFile).ToList();
                 _logger.LogDebug($"Found in total {nugetReferences.Count} references of NuGets (direct & transient)");
 
                 var usages = _usageFinder.FindUsagesOf(nugetReferences, packagesToRetire);
                 if (usages.Any())
                 {
                     foundVulnerabilities = true;
-                    var plural = usages.Count > 1 ? "s" : string.Empty;
-                    var grouped = usages.GroupBy(g => g.NugetReference.ToString());
+
+                    var project = new Project
+                    {
+                        Name = lockFile.PackageSpec.Name,
+                        Path = lockFile.PackageSpec.FilePath,
+                    };
+                    report.Add(project);
+
+                    var grouped = usages.GroupBy(g => g.NugetReference.ToString()).ToList();
+                    var usagePlural = usages.Count > 1 ? "s" : string.Empty;
+                    var groupPlural = grouped.Count > 1 ? "s" : string.Empty;
                     var sb = new StringBuilder();
-                    sb.AppendLine($"Found use of {grouped.Count()} vulnerable libs in {usages.Count} dependency path{plural}.");
+                    sb.AppendLine($"Found use of {grouped.Count} vulnerable lib{groupPlural} in {usages.Count} dependency path{usagePlural}.");
 
                     foreach (var group in grouped)
                     {
+                        var firstGroup = group.FirstOrDefault();
+                        var keyPieces = group.Key.Split('/');
+                        var issue = new ProjectIssue
+                        {
+                            // Name = keyPieces[0],
+                            // Version = keyPieces[1],
+                            Description = firstGroup?.Description,
+                            IssueUrl = firstGroup?.IssueUrl.ToString(),
+                            ProblemPackage = firstGroup?.GetPackageChain().LastOrDefault(),
+                            PackageChains = group.Select(g => g.GetPackageChain()).Distinct().ToList(),
+                        };
+                        project.Issues.Add(issue);
                         sb.AppendLine();
-                        sb.AppendLine($"* {group.FirstOrDefault().Description} in {group.Key.Red()}");
-                        sb.AppendLine(group.FirstOrDefault().IssueUrl.ToString());
+                        sb.AppendLine($"* {firstGroup?.Description} in {group.Key.Red()}");
+                        sb.AppendLine(firstGroup?.IssueUrl.ToString());
 
-                        foreach (var usage in group.Where(x => !x.IsDirect))
+                        // looks like occasionally the test is seeing obscene numbers of dep paths...
+                        // Found use of 4 vulnerable libs in 10990 dependency paths
+                        // let's guard against that.
```

Comment:
For what reason (what's the issue this is solving)?

Reply:
I saw some docker test runs that ran for like 5/6 minutes, and failed after timing out. They had this walllllllllllll of repeating dependency chains. I don't know what possessed me to suss out that it actually found 10990 dependency paths, but the tests ran a number of times after that was in place and I didn't run into it again, so it's doing the job at least... :D